### PR TITLE
Optimizing response read

### PR DESCRIPTION
**Pull request by amitkanfer**

### CODE REVIEW

File: libbeat/esleg/eslegclient/connection.go [hunk 1]
```diff
@@ -452,7 +452,9 @@ func (conn *Connection) execHTTPRequest(req *http.Request) (int, []byte, error)
 	defer closing(resp.Body, conn.log)
 
 	status := resp.StatusCode
-	obj, err := ioutil.ReadAll(resp.Body)
+	obj := new(bytes.Buffer)
+	_, err = io.Copy(obj, resp.Body)
+	// obj, err := ioutil.ReadAll(resp.Body)
```
Comment (belimawr) [inline libbeat/esleg/eslegclient/connection.go:457]: You can remove this line.
```suggestion
```

**Review by belimawr — COMMENTED** [review]
@amitkanfer I'm a bit confused about what you mean by "optimising". Did you see a performance bottleneck on `ioutil.ReadAll` that is solved by using a `bytes.Buffer` and `io.Copy`? Or is it just getting rid of the deprecated `ioutil.ReadAll`?

File: libbeat/esleg/eslegclient/connection.go (hunk 1 above)
Comment (aleksmaus) [reply libbeat/esleg/eslegclient/connection.go:457]: nits:
Would be nice to benchmark.
Also, could possibly check the ContentLength if available and preallocate the buffer to the size.
```
if resp.ContentLength > 0 {
    buf = bytes.NewBuffer(make([]byte, 0, resp.ContentLength))
```

Comment (rdner) [reply libbeat/esleg/eslegclient/connection.go:457]: @aleksmaus see https://github.com/elastic/beats/issues/36151#issuecomment-1931696767

**Pull request closed without merging**